src/meta/src/stream/source_manager.rs

```diff
@@ -778,6 +873,63 @@ impl SourceManager {
         Ok(assigned)
     }
 
+    pub async fn allocate_splits_for_backfill(
```

Comment:
Does it need `migrate_splits_for_backfill` as well? May need to test upon split repartition.

Reply:
BTW, please add some rustdocs ❤️

Reply:
> Does it need migrate_splits_for_backfill as well? 

Yes. Reschedule is not fully considered yet. 

> May need to test upon split repartition.

Indeed. BTW, split change is handled by `reassign_splits`. Currently we have no tests for that at all. 😕 Will try to add some.